### PR TITLE
Add hyper parameters to FireworksSFTConfig and FireworksFineTuningRequest

### PR DESCRIPTION
**Pull request by anndvision**
1. Added new parameters to all relevant structs:
    - early_stop: Whether to early stop the fine-tuning job
    - epochs (usize/int): The number of epochs to use for the fine-tuning job
    - max_context_length (usize/int): The maximum context length to use for the fine-tuning job
    - learning_rate (f64/float): The learning rate to use for the fine-tuning job
    - lora_rank (usize/int): The rank of the LoRA matrix to use for the fine-tuning job
    - batch_size (usize/int): The batch size to use for the fine-tuning job (tokens)
    - display_name: Display name for the fine-tuning job
    - output_model: Model ID for the resulting fine-tuned model
    - warm_start_from: PEFT addon model to fine-tune from
    - is_turbo: Enable turbo mode
    - eval_auto_carveout: Auto-carve dataset for evaluation
    - nodes: Number of nodes for the job
    - mtp_enabled: Enable Multi-Token Prediction
    - mtp_num_draft_tokens: Number of draft tokens for MTP
    - mtp_freeze_base_model: Freeze base model for MTP
  1. Updated files:
    - Core structs (FireworksFineTuningRequest, FireworksSFTConfig, UninitializedFireworksSFTConfig)
    - Python bindings (new and __init__ methods)
    - Load method to transfer parameters
    - Optimizer implementation to include parameters in the request
    - Test file with all new parameters
    - Python type stub file (.pyi)
    - TypeScript bindings (auto-generated)
    - Documentation/docstrings for all new parameters

----

> [!IMPORTANT]
> Add new hyperparameters to `FireworksSFTConfig` and related structures, updating Python, TypeScript, and Rust implementations, including tests and UI adjustments.
> 
>   - **Behavior**:
>     - Added new hyperparameters to `FireworksSFTConfig`, `FireworksFineTuningRequest`, and `UninitializedFireworksSFTConfig`.
>     - Parameters include `early_stop`, `epochs`, `max_context_length`, `learning_rate`, `lora_rank`, `batch_size`, `display_name`, `output_model`, `warm_start_from`, `is_turbo`, `eval_auto_carveout`, `nodes`, `mtp_enabled`, `mtp_num_draft_tokens`, `mtp_freeze_base_model`.
>   - **Bindings**:
>     - Updated Python bindings in `tensorzero.pyi` for `FireworksSFTConfig`.
>     - Updated TypeScript bindings in `FireworksSFTConfig.ts` and `UninitializedFireworksSFTConfig.ts`.
>   - **Core**:
>     - Modified `mod.rs` to include new parameters in `FireworksFineTuningRequest` and `FireworksSFTConfig`.
>     - Updated `load` method to transfer new parameters.
>   - **Tests**:
>     - Added tests for new parameters in `test_optimization.py` and `fireworks_sft.rs`.
>   - **UI**:
>     - Updated `client.ts` to handle new parameters in `launch_sft_job` function.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=tensorzero%2Ftensorzero&utm_source=github&utm_medium=referral)<sup> for a98bc6e770671b5a3df59068f4e16225ec24e1a8. You can [customize](https://app.ellipsis.dev/tensorzero/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: ui/app/utils/supervised_fine_tuning/client.ts [hunk 1]
```diff
@@ -55,6 +55,21 @@ export async function launch_sft_job(
     optimizerConfig = {
       type: "fireworks_sft",
       model: data.model.name,
+      early_stop: null,
+      epochs: 1,
+      learning_rate: 0.0001,
```
Comment (virajmehta) [inline ui/app/utils/supervised_fine_tuning/client.ts:60]: do we want to hardcode these values?

Comment (anndvision) [reply ui/app/utils/supervised_fine_tuning/client.ts]: set to null

**Review by virajmehta — APPROVED** [review]
